### PR TITLE
Simplify Status class.

### LINKED ISSUE #1813: Complete implementation of google::cloud::Status.
Or close to complete, I think we need the following changes:

* The `status_code()` member function becomes `code()` and returns `StatusCode` instead of `long`.
* The `error_message()` member function becomes `message()`.
* Remove the `error_details` stuff.
* Remove the `operator<` and friends.
* Change `operator<<(std::ostream& os, Status&)` to represent the error codes as human readable strings.

### PR DESCRIPTION
**Pull request by coryan**
Make it more similar to the internal version.

This fixes #1813.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1842)

### CODE REVIEW

File: google/cloud/status.h [hunk 1]
```diff
@@ -60,36 +60,31 @@ std::ostream& operator<<(std::ostream& os, StatusCode code);
...
  * This class is modeled after `grpc::Status`, it contains the status code and
  * error message (if applicable) from a JSON request.
- *
- * TODO(#548) - define well-known error codes.
  */
 class Status {
  public:
   Status() : status_code_(StatusCode::kOk) {}
 
   explicit Status(StatusCode status_code, std::string error_message)
```
Comment (devbww) [inline google/cloud/status.h]: s/error_message/message/?

Comment (coryan) [reply google/cloud/status.h]: Done.

File: google/cloud/status.h [hunk 2]
```diff
@@ -60,36 +60,31 @@ std::ostream& operator<<(std::ostream& os, StatusCode code);
...
   }
   bool operator!=(Status const& rhs) const { return not(*this == rhs); }
 
   StatusCode code() const { return status_code_; }
-  long status_code() const { return status_code_; }
-  std::string const& error_message() const { return error_message_; }
+  std::string const& message() const { return message_; }
 
  private:
   StatusCode status_code_;
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/status.h]: Done.

**Comment by coryan** [comment]
PTAL.

**Pull request merged**